### PR TITLE
[Stunner] Basic Tasks Tests

### PR DESCRIPTION
**Pull request by LuboTerifaj**
(Un)Marshaller tests for Basic Tasks:

- None Tasks

- Script Tasks

- Business Rule Tasks

- User Tasks

### CODE REVIEW

**Comment by LuboTerifaj** [comment]
Hi @evacchi, @hasys 
I have created new (un)marshaller tests for Basic Tasks. Can you please have a look at this PR?
Thank you!

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/marshalling/tasks/UserTaskTest.java [hunk 1]
```diff
@@ -50,4 +102,1081 @@ public void RHBA607() throws Exception {
...
+        final String TASK_SUBJECT_MVEL = "subject07 ~!@#$%^&*()_+`-={}|[]\\:\";'<>?,./";
+        final String TASK_ACTORS_MVEL = "customUser";
+        final String TASK_GROUPS_MVEL = "customGroup";
+        final String TASK_PRIORITY_MVEL = "10";
+        final String TASK_DESCRIPTION_MVEL = "Task07 description\n ~!@#$%^&*()_+`1234567890-={}|[]\\:\";'<>?,./";
+        final String TASK_CREATED_BY_MVEL = "customUser";
+        final String TASK_ON_ENTRY_ACTION_MVEL = "System.out.println(\"On Entry Action\");";
+        final String TASK_ON_EXIT_ACTION_MVEL = "System.out.println(\"On Exit Action\");";
+
+        final String TASK_DATA_INPUT_OUTPUT_JAVA = "|input:String,GroupId:Object,Skippable:Object,Comment:Object,Description:Object,Priority:Object,CreatedBy:Object||output:String|[din]processGlobalVar->input,[dout]output->processGlobalVar";
```
Comment (LuboTerifaj) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/marshalling/tasks/UserTaskTest.java]: Just want to ask. Does anyone know, why this value is different in task with name "Task01 name..."?

**Comment by LuboTerifaj** [comment]
@evacchi @hasys 
Yes, some of them are still failing, but this is known issue, isn't it? What am I supposed to report? Isn't it already reported? Thanks!

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/marshalling/tasks/Task.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    static final boolean HAS_OUTCOME_EDGE = true;
+    static final boolean HAS_NO_OUTCOME_EDGE = false;
+
+    protected DiagramMarshaller<Graph, Metadata, Diagram<Graph, Metadata>> marshaller = null;
+
+    @Parameterized.Parameters
+    public static List<Object[]> marshallers() {
+        return Arrays.asList(new Object[][]{
+                // New (un)marshaller is disabled for now due to found incompleteness
+                {OLD}//, {NEW}
```
[the remark this answers is not in the document]
Comment (LuboTerifaj) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/marshalling/tasks/Task.java]: @evacchi yes, some tests are still failing.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/marshalling/tasks/UserTaskTest.java (hunk 1 above)
Comment (LuboTerifaj) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/marshalling/tasks/UserTaskTest.java]: Hi @evacchi
Do you know, why there is different Input/Output only in one task within the whole process, where are many tasks? Thank you.

**Comment by evacchi** [comment]
@LuboTerifaj open a new Jira to report that these tests are broken under the new Marshallers

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/marshalling/tasks/UserTaskTest.java (hunk 1 above)
Comment (evacchi) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/marshalling/tasks/UserTaskTest.java]: sorry I don't understand the question

**Comment by LuboTerifaj** [comment]
Hi @evacchi 
Since issue is already reported [1], it can be merged yes?
Thank you!
[1] https://issues.jboss.org/browse/JBPM-7259

**Comment by romartin** [comment]
hey @LuboTerifaj ,
Sorry for late review of this. Now it needs rebasing, lemme know please when I can take a look again.
Thanks!

**Comment by LuboTerifaj** [comment]
Hi @romartin 
It should be OK now.
Thanks.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/marshalling/tasks/UserTaskTest.java (hunk 1 above)
Comment (LuboTerifaj) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/marshalling/tasks/UserTaskTest.java]: In the whole process are 24 tasks. I have set for each task the same input and output data. If you call `executionSet.getAssignmentsinfo().getValue()`, where `executionSet` is of type `UserTaskExecutionSet`, you get for all tasks this String: `|input:String,Skippable:Object,GroupId:Object,Comment:Object,Description:Object,Priority:Object,CreatedBy:Object||output:String|[din]processGlobalVar->input,[dout]output->processGlobalVar` except one task for which you get different value as you can see on commented line. Do you know, why it differs? Thank you.

Comment (evacchi) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/marshalling/tasks/UserTaskTest.java]: unfortunately I don't... these internal inconsistencies are something I have struggled with while implementing the new marshallers. Maybe @tsurdilo or @wmedvede know better

Comment (wmedvede) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/marshalling/tasks/UserTaskTest.java]: Same here, no idea. Unfortunately I'm not from the epoch of all this legacy code.

**Pull request merged**